### PR TITLE
[Feature] CON 81 - Enable text wrap passages

### PR DESCRIPTION
**Pull request by habemuscode**
**Related to this issue**

https://oat-sa.atlassian.net/browse/CON-81

**Other related PRs**

- https://github.com/oat-sa/extension-tao-itemqti/pull/1580
- https://github.com/oat-sa/extension-tao-mediamanager/pull/240

**Story**

As the author, I want to be able to wrap my text around pictures, so that I can produce passages that look like real-life media content.

Currently, I am only able to anchor images AS a character. In order for the text to be able to flow around it, I will need to be able to anchor it TO a character and specify how text is supposed to wrap around it

Required wrap modes:

- Float off
- Float right
- Float left

**Acceptance Criteria**

- Anchoring an image as a character should still be possible
- Text wrapping should not be available when the image is anchored as a character
- Images maintain position within their paragraph when text is added before or after
- Responsive images are still supported if the image is anchored to a character
- Exported items will contain the relevant CSS to ensure the rendering will be correct in any environment
- Exporting and reimporting an item will maintain the anchoring and wrapping

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -63,7 +63,7 @@
     'label' => 'TAO Base',
     'description' => 'TAO meta-extension',
     'license' => 'GPL-2.0',
-    'version' => '46.8.1',
+    'version' => '46.8.2',
```
Comment (gabrielfs7) [inline manifest.php:66]: Should not be 46.9.0 since it is a feature?

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: manifest.php (hunk 1 above)
Comment (habemuscode) [reply manifest.php:66]: You are right, pushed

**Review by gabrielfs7 — APPROVED** [review]
LGTM Good job. BE part is OK.

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Review by boajer — APPROVED** [review]
Backend part looks good for me.

Manifest update is missing.

File: views/scss/inc/_action-bars.scss [hunk 2]
```diff
@@ -51,6 +51,21 @@
                 padding-right: 34px;
                 min-width: 250px !important;
             }
+            .search-area-buttons-container {
```
Comment (bziondik) [inline views/scss/inc/_action-bars.scss:54]: Why you update styles for search-area?

File: views/scss/inc/_action-bars.scss [hunk 3]
```diff
@@ -209,11 +224,14 @@
             .search-area {
                 float: none;
                 position: absolute;
-                right: 0px;
+                right: 0;
```
Comment (bziondik) [inline views/scss/inc/_action-bars.scss:227]: Can it be some regression after merge develop?

**Review by bziondik — CHANGES_REQUESTED** [review]
Check changes with search-area, looks like some regression after merge develop

File: views/scss/inc/_action-bars.scss (hunk 2 above)
Comment (habemuscode) [reply views/scss/inc/_action-bars.scss:54]: Looks like the fixes of @404passionFruit. Are good or not?

File: views/scss/inc/_action-bars.scss (hunk 3 above)
Comment (habemuscode) [reply views/scss/inc/_action-bars.scss:227]: @404passionFruit. Are good or not?

File: views/scss/inc/_action-bars.scss (hunk 2 above)
Comment (404passionFruit) [reply views/scss/inc/_action-bars.scss:54]: It's weird, cause those changes come from advanced search. https://github.com/oat-sa/tao-core/blob/feature/TAO-10203-advanced-search/views/scss/inc/_action-bars.scss#L52 

Do you have any idea how it got here?

Comment (404passionFruit) [reply views/scss/inc/_action-bars.scss:54]: From [history](https://github.com/oat-sa/tao-core/commits/dab1d9e2cf91285018c7b19737fcad4193994b5a/views/scss/inc/_action-bars.scss) I see 2 commits dated Dec 17 that add those changes. Could it be some copy/paste mistake from advanced search environment?

File: manifest.php [hunk 4]
```diff
@@ -63,7 +63,7 @@
     'label' => 'TAO Base',
     'description' => 'TAO meta-extension',
     'license' => 'GPL-2.0',
-    'version' => '46.8.1',
+    'version' => '46.9.0',
```
Comment (404passionFruit) [inline manifest.php]: Do we need version bump?

File: views/package.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
         "@oat-sa/tao-core-libs": "0.4.2",
         "@oat-sa/tao-core-sdk": "1.7.0",
         "@oat-sa/tao-core-shared-libs": "1.0.3",
-        "@oat-sa/tao-core-ui": "1.19.1",
+        "@oat-sa/tao-core-ui": "git://github.com/oat-sa/tao-core-ui-fe#feature/CON-81/enable-text-wrap-passages",
```
Comment (404passionFruit) [inline views/package.json]: Please change it to the version number

File: views/scss/inc/_action-bars.scss (hunk 2 above)
Comment (404passionFruit) [reply views/scss/inc/_action-bars.scss:54]: Anyway I don't see a point in adding those changes. Your base branch is develop, so if it's not related to the issue you are working on, it seems like a good idea to stick to what's in master & develop branches. I would suggest to revert those changes.

File: views/scss/inc/_action-bars.scss (hunk 3 above)
Comment (404passionFruit) [reply views/scss/inc/_action-bars.scss:227]: Seems like this file is not related to the issue at all, we should revert the changes in it.

Comment (404passionFruit) [reply views/scss/inc/_action-bars.scss:227]: Good catch, @bziondik!

**Review by 404passionFruit — CHANGES_REQUESTED** [review]
Please have a look at the comments. Seems like some scss changes should be reverted and bundles should be regenerated.

**Comment by habemuscode** [comment]
I will create PR again

**Comment by habemuscode** [comment]
New PR: https://github.com/oat-sa/tao-core/pull/2804

**Pull request closed without merging**